src/main/webapp/app/shared/components/advancedSearchModal/advancedSearchModal.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Modal, ModalHeader, ModalBody, Label, Input } from 'reactstrap';
+import { AvFeedback, AvForm, AvGroup, AvInput, AvField } from 'availity-reactstrap-validation';
+import { Translate } from 'react-jhipster';
+import { IRootState } from 'app/shared/reducers';
+import { getYearSessionOptions } from 'app/entities/user-cc-info/user-cc-info.reducer';
+import { getCourseProgramOptions } from 'app/entities/user-uni-info/user-uni-info.reducer';
+import { IFilter, setEntity } from './advancedSearchModal.reducer';
+
+import './advancedSearchModal.scss';
+
+interface IFilterSearchBarProps extends StateProps, DispatchProps {
+  isOpen: boolean;
+  toggleModal: () => void;
+  familyCode: string;
+  searchUsers: (familyCode: string, filters: any) => void;
+}
+
+interface IAdvancedSearchModalState {
+  filters: IFilter;
+}
+
+class AdvancedSearchModal extends React.Component<IFilterSearchBarProps, IAdvancedSearchModalState> {
+  constructor(props: IFilterSearchBarProps) {
+    super(props);
+    this.state = {
+      filters: {
+        userFirstName: '',
+        userLastName: '',
+        courseProgramId: undefined,
+        yearSession: '',
+      },
+    };
+  }
+
+  componentDidUpdate(prevProps: IFilterSearchBarProps) {
+    if (this.props.isOpen !== prevProps.isOpen) {
+      this.props.getYearSessionOptions(0, 12, 'value,desc');
+      this.props.getCourseProgramOptions(0, 12, 'value,desc');
+    }
+  }
+
+  renderYearSessionOptions = () => this.props.yearSessionOptions.map(year => <option key={year}>{year}</option>);
```

Comment:
option no set value will default to?

Reply:
It will not filter members with enrolled year. Eg: if there are multiple members' name consisted of Lee, without selecting the enrolled year filter when searching for Lee, it will then return all the Lee from all enrolled year.

Reply:
@junzai97 i mean the option tag value attribute
Usually we will set the value attribute right?

Reply:
If there is no value attribute set, it will take the value as the text in between the option tag. But practice wise, I think it better for us to have the value attribute as well. So this has been resolved in 7b4fa986380f7e4111e7ba54bece5fb2ec91270f